var/spack/repos/builtin/packages/eccodes/package.py

```diff
@@ -0,0 +1,87 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class Eccodes(CMakePackage):
+    """ecCodes is a package developed by ECMWF which provides an application
+    programming interface and a set of tools for decoding and encoding
+    messages in the following formats:
+
+      WMO FM-92 GRIB edition 1 and edition 2
+      WMO FM-94 BUFR edition 3 and edition 4 
+      WMO GTS abbreviated header (only decoding).
+    """
+
+    homepage = "https://software.ecmwf.int/wiki/display/ECC/ecCodes+Home"
+    url      = "https://software.ecmwf.int/wiki/download/attachments/45757960/eccodes-2.2.0-Source.tar.gz?api=v2"
+
+    version('2.2.0', 'b27e6f0a3eea5b92dac37372e4c45a62')
+
+    variant('netcdf', default=True)
+    variant('openjpeg', default=True)
+    variant('libpng', default=True)
+    variant('python', default=False)
+    variant('pthreads', default=False)
+    variant('openmp', default=False)
+    variant('memfs', default=False)
+
+    depends_on('netcdf', when='+netcdf')
+    depends_on('openjpeg', when='+openjpeg')
+    depends_on('libpng', when='+libpng')
+    extends('python', when='+python')
+
+    def cmake_args(self):
```

Comment:
This could be made a LOT more succinct (and clearer).  For example:
```
    def cmake_args(self):
        spec = self.spec
        return [
            '-DUSE_EVERYTRACE=%s' % ('YES' if '+everytrace' in spec else 'NO'),
            '-DBUILD_PYTHON=%s' % ('YES' if '+python' in spec else 'NO'),
            '-DBUILD_GRIDGEN=%s' % ('YES' if '+gridgen' in spec else 'NO'),
            '-DBUILD_COUPLER=%s' % ('YES' if '+coupler' in spec else 'NO'),
            '-DBUILD_MODELE=%s' % ('YES' if '+modele' in spec else 'NO'),
            '-DUSE_PISM=%s' % ('YES' if '+pism' in spec else 'NO'),
            '-DBUILD_DOCS=%s' % ('YES' if '+doc' in spec else 'NO')]
```
There are other ways to be even more succinct if you wish to use a helper function.